user/languages/dart.md

```diff
@@ -16,66 +16,137 @@ This guide covers build environment and configuration topics specific to
 ```
 
-## Default Dart Version
+This creates a separate Travis build for each Dart version. It can be used in
+conjunction with `env` or similar fields to create a [build matrix][].
 
-If you leave the `dart:` key out of your `.travis.yml`, Travis CI will use
-the latest release in the `stable` channel.
+[build matrix]: /user/customizing-the-build/#Build-Matrix
 
 ## Dependency Management
 
-If your Dart package has a `pubspec.yaml` file, then `pub get` will be executed
-to install any dependencies of the package.
+If your Dart package has a `pubspec.yaml` file, then `pub get` will be run
+before your tests to install any dependencies of the package.
+
+## Running Tests
+
+If your package depends on the [`test` package][test], `pub run test` will be
+run by default. This typically only runs tests on the Dart VM, but you can
+[configure it][] to run on additional platforms by default.
+
+[test]: https://pub.dartlang.org/packages/test
+[configure it]: https://github.com/dart-lang/test/blob/master/doc/configuration.md#platforms
+
+You can also customize the arguments Travis passes to the test runner using the
+`dart_tasks` field in `.travis.yml`.
+
+```yaml
+language: dart
+dart_tasks:
+- test: --platform vm
+- test: --platform chrome
+```
+
+Each task creates a separate Travis build. It can be used in conjunction with
+`env`, `dart`, or similar fields to create a [build matrix][].
+
+### Available Browsers
+
+Travis comes with Firefox and Chrome installed by default on Linux, and Safari
+on OS X. However, if you want to run your tests on [Dartium][], you'll need to
+install it by adding `install_dartium: true` either at the top level or for a
+particular task.
+
+[Dartium]: https://webdev.dartlang.org/tools/dartium
+
+```yaml
+language: dart
+dart_tasks:
+- test: --platform vm
+- test: --platform dartium
+  install_dartium: true
+```
+
+### XVFB
+
+On Linux, the test runner uses [XVFB][] by default so that it can use browsers
+like Chrome that require a display. However, this may interfere with certain
+applications, so you can turn it off by setting `xvfb: false` either at the top
+level or for a particular task.
 
-## Default Test Script
+[XVFB]: https://www.x.org/archive/X11R7.6/doc/man/man1/Xvfb.1.xhtml
 
-### Tests written with test package
+```yaml
+language: dart
+dart_tasks:
+- test: --exclude-tags no-xvfb
+- test: --tags no-xvfb
+  xvfb: false
+```
 
-The tests are done by the
-[test package](https://pub.dartlang.org/packages/test). This tool
-will automatically detect and run all the tests in your Dart project in the
-correct environment.
+XVFB is never used on OS X, since it doesn't use the X windows system.
 
-Only *VM* tests are run by default. If you set
-`with_content_shell: true` in your `.travis.yml` then the tests for platforms
-`vm`, `content-shell` and `firefox` will be run.
+## Other Tasks
 
-### Tests written with unittest package (deprecated)
+Several tasks are available in addition to running tests.
```

Comment:
This is under the "Other Tasks" header. It's meant as a brief introductory sentence for the sections below on specific tasks.